docs/tutorials/control/flutter-app.md

```diff
@@ -0,0 +1,732 @@
+      _locations = await _viam.appClient.listLocations(_organization);
+
+      // in Flutter setState tells the UI to rebuild the widgets whose state has changed,
+      // this is how we change from showing a loading screen to a list of values
+      setState(() {
+        _loading = false;
+      });
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  /// This method will navigate to a specific [Location].
+  void _navigateToLocation(Location location) {
+    // Navigator.of(context)
+    //     .push(MaterialPageRoute(builder: (_) => LocationScreen(_viam, location)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Locations')),
+      // If we are loading, show a loading indicator.
+      // Otherwise, show a list of [Locations]s.
+      body: _loading
+          ? Center(
+              child: const CircularProgressIndicator.adaptive(),
+            )
+          : // Build a list from the [_locations] state.
+          ListView.builder(
+              itemCount: _locations.length,
+              itemBuilder: (_, index) {
+                final location = _locations[index];
+                return ListTile(
+                  title: Text(location.name),
+                  onTap: () => _navigateToLocation(location),
+                  trailing: const Icon(Icons.chevron_right),
+                );
+              },
+            ),
+    );
+  }
+}
+```
+
+### Get the Viam API key
+
+Notice in the file the following line:
+
+```{class="line-numbers linkable-line-numbers"}
+_viam = await Viam.withApiKey(dotenv.env['API_KEY_ID']?? '', dotenv.env['API_KEY']?? '');
+```
+
+You need to get the API keys from the Viam app and store them in an environmental variables file to so your mobile app can access your smart machine:
+
+1. In your project folder, create a file to store your API keys.
+   Name it <file>.env</file>.
+   Copy and paste these two lines into the file:
+
+   ```{class="line-numbers linkable-line-numbers"}
+   API_KEY_ID="PASTE YOUR API KEY ID HERE"
+   API_KEY="PASTE YOUR API KEY HERE"
+   ```
+
+2. Go to the [Viam app](https://app.viam.com) and log in.
+
+3. Click the organization dropdown menu on the right side of the top banner.
+   If you're not already in the organization you want to connect to, click the correct organization name to navigate to it.
+
+   {{<imgproc src="/tutorials/flutter-app/org-dropdown.png" resize="1200x" style="max-width:800px" declared-dimensions="true" alt="The Viam web app with the organization dropdown labeled 'Clint's Org' highlighted.">}}
+
+4. Click the organization dropdown menu again and click **Settings**.
+
+5. Scroll to the **API Keys** section.
+   You can find and use an existing API key for your smart machine, or you can create a new one for this application.
+   For this tutorial we will create a new one:
+
+   1. Click **Generate key**.
+   2. Give the key a name like "flutter-app-my-org-name."
```

Comment:
```suggestion
   2. Give the key a name like "flutter-app-my-org-name".
```

Oh wait, this is a thing, quotation marks and periods. I have no preference, ignore me, don't shoot.

Reply:
Lol wait I caught this earlier and changed it to be period then quotation mark, since that's correct right? If it was monospace that'd be a different story, but since it isn't monospace I should leave as-is right? With period then quotation?